tests/test_enterprise/api/test_default_enrollment_views.py

```diff
@@ -0,0 +1,726 @@
+        assert response_data['detail'] == (
+            f'User with lms_user_id {self.user.id} is not associated with '
+            f'the enterprise customer {str(self.enterprise_customer.uuid)}.'
+        )
+
+    @mock.patch('enterprise.content_metadata.api.EnterpriseCatalogApiClient')
+    @mock.patch('enterprise.models.utils.get_best_mode_from_course_key')
+    def test_default_enterprise_enrollment_intentions_learner_status_enrollable(
+        self,
+        mock_get_best_mode_from_course_key,
+        mock_catalog_api_client,
+    ):
+        """
+        Test default enterprise enrollment intentions for specific learner linked to enterprise customer, where
+        the course run associated with the default enrollment intention is enrollable.
+        """
+        self.set_jwt_cookie(ENTERPRISE_LEARNER_ROLE, str(self.enterprise_customer.uuid))
+        enrollment_intention = create_mock_default_enterprise_enrollment_intention(
+            enterprise_customer=self.enterprise_customer,
+            mock_catalog_api_client=mock_catalog_api_client,
+        )
+        mock_get_best_mode_from_course_key.return_value = VERIFIED_COURSE_MODE
+        factories.EnterpriseCustomerUserFactory(
+            user_id=self.user.id,
+            enterprise_customer=self.enterprise_customer,
+        )
+        query_params = f'enterprise_customer_uuid={str(self.enterprise_customer.uuid)}'
+        response = self.client.get(
+            f"{settings.TEST_SERVER}{DEFAULT_ENTERPRISE_ENROLLMENT_INTENTION_LEARNER_STATUS_ENDPOINT}?{query_params}"
+        )
+        assert response.status_code == status.HTTP_200_OK
+        response_data = response.json()
+        assert response_data['lms_user_id'] == self.user.id
+        assert response_data['user_email'] == self.user.email
+        assert response_data['enrollment_statuses'] == {
+            'needs_enrollment': {
+                'enrollable': [
+                    self.get_default_enrollment_intention_with_learner_enrollment_state(enrollment_intention)
+                ],
+                'not_enrollable': [],
+            },
+            'already_enrolled': [],
+        }
+        assert response_data['metadata'] == {
+            'total_default_enterprise_enrollment_intentions': 1,
+            'total_needs_enrollment': {
+                'enrollable': 1,
+                'not_enrollable': 0,
+            },
+            'total_already_enrolled': 0,
+        }
+
+    @ddt.data(
+        {'run_is_enrollable': False, 'unenrollment_exists': False},
+        {'run_is_enrollable': True, 'unenrollment_exists': True},
+    )
+    @mock.patch('enterprise.content_metadata.api.EnterpriseCatalogApiClient')
+    @mock.patch('enterprise.models.utils.get_best_mode_from_course_key')
+    @mock.patch.object(EnterpriseCourseEnrollment, 'course_enrollment', new_callable=mock.PropertyMock)
+    @ddt.unpack
+    def test_default_enrollment_intentions_learner_status_content_not_enrollable(
```

Comment:
This is the only changed test, everything else has just been refactored into this new file.

Reply:
Good call to start breaking up the giant `test_views` file :)